### PR TITLE
Ongoing OIDC: implement the token request capability in the repo.

### PR DESCRIPTION
**Pull request by jimlambrt**
This PR adds:
* oidc.TokenRequest(...) which accepts a token request id from the Boundary client and either issues a token or returns and error
* authtoken.(Repository).IssuePendingToken which either issues a pending token or returns and error.

### CODE REVIEW

File: internal/authtoken/repository.go [hunk 1]
```diff
@@ -361,6 +361,55 @@ func (r *Repository) DeleteAuthToken(ctx context.Context, id string, opt ...Opti
 // "issued".  If the token has already been issued, an error is returned with a
 // nil token.  If no token is found for the tokenRequestId an error is returned
 // with a nil token.
+//
+// Note: no oplog entries are created for auth token operations (this is intentional).
 func (r *Repository) IssueAuthToken(ctx context.Context, tokenRequestId string) (*AuthToken, error) {
-	panic("to-do")
+	const op = "authtoken.(Repository).IssueAuthToken"
```
Comment (talanknight) [inline internal/authtoken/repository.go:367]: Now that issuing the auth token isn't done at the same time as creation of the auth token, do we need to update the auth token expiration time or last accessed time based on the issue time?

**Review by talanknight — APPROVED** [review]

File: internal/authtoken/repository.go (hunk 1 above)
Comment (jimlambrt) [reply internal/authtoken/repository.go:367]: I had asked about expiry earlier, and we sort of aligned on need need to update it when it's issued.   Last time accessed should be updated I guess.  I'm thinking, there should be some trigger that bumps the timestamp when the token is updated to issued. Make sense?

File: internal/authtoken/repository.go [hunk 2]
```diff
@@ -361,6 +361,55 @@ func (r *Repository) DeleteAuthToken(ctx context.Context, id string, opt ...Opti
...
+			if err != nil {
+				return errors.Wrap(err, op)
+			}
+			if rowsUpdated == 0 {
+				return errors.New(errors.RecordNotFound, op, "pending auth token not found")
+			}
+			if rowsUpdated > 1 {
+				return errors.New(errors.Internal, op, fmt.Sprintf("should have updated 1 row and we attempted to update %d rows", rowsUpdated))
+			}
+			if at.Status != string(IssuedStatus) {
```
[the remark this answers is not in the document]
Comment (jimlambrt) [reply internal/authtoken/repository.go:392]: Currently, there is no ticker running to update these to some other status when they expire in a pending status.  I'll add that as a TODO and open another PR for that.

[the remark this answers is not in the document]
Comment (jimlambrt) [reply internal/authtoken/repository.go:392]: ICU-1250

**Review by talanknight — APPROVED** [review]

**Pull request merged**